### ISSUE TITLE
Accessibility: Create consistent nested header order

### ISSUE THREAD

**Issue by aparkening** [alistapart/AListApart#216]
From Andrew's accessibility audit:
Heading nesting order is not regular – headings are a bit all over the place. (arguably a WCAG 2.0 issue, but ripe for disagreement)

- In the footer for the home page, for example, the “more from ALA” section is an H2, which contains an H2, which contains an H1.  And then to the right there are a couple of H4’s thrown in.  It doesn’t seem like the presentation matches the semantics. 

- The order for headings needs to be adjusted throughout the site. For example, in Articles the heading levels are funky.  For example, “illustration” and “share this” shouldn’t be H1 headings.
![header-order](https://cloud.githubusercontent.com/assets/7241111/7903592/829eea88-07af-11e5-9ced-223ce87e4bb2.png)

**Comment by murtaugh** [comment]
@awkawk I'd love some assistance thinking about this. 

The header orders were developed to make sense within the HTML5 spec (namely, you can start over with H1s whenever you enter a new sectioning element). (There are probably cases where we've mis-nested, but that's the overall intent.)

This approach allows the HTML to be more modular (not having to concern ourselves with header levels across multiple templates is a blessing), but does this does not work with accessibility tools?

How much of an issue is this, within the context of accessibility? It would not be a small task to re-factor all the headers, and would result in (what I consider to be) undesirable mostly-redundant templates. Are there approaches we might consider?

**Comment by aparkening** [comment]
I've pinged Andrew about this via email today.

**Comment by awkawk** [comment]
Sorry for the delay.

I wish that we could rely on better support with user agents for the HTML5 outlining rules but it isn't there yet.  See http://www.w3.org/TR/html/sections.html#headings-and-sections and in particular the note at 4.3.10.1 - essentially the advice is to keep doing headings as in the past.

WCAG has a concept called "accessibility support" which basically says that you can't rely on what a spec says unless there is some evidence of implementation that end-users can take advantage of.

I'm sure that the fix is ugly and results in extra templates, but wonder if there is another way.  Do you think that it is possible to add aria-role and aria-level so that we can reassign the heading to be correct, or is that not possible from the templates?  I'll do a quick test on that here to make sure it is working as expected, but that might give us the improvement without all of the re-work?

**Comment by awkawk** [comment]
As I think about this more it is an issue that the "first rule of ARIA" is to not override native semantics, which we are doing if we change h3s to h2s, although it is clearly less egregious than changing an h2 into a button or something like that. on the fly script modification of elements is possible, but probably not ideal.

**Comment by murtaugh** [comment]
>  Do you think that it is possible to add aria-role and aria-level so that we can reassign the heading to be correct, or is that not possible from the templates?

That would have the same issue. We could pass in parameters to tell the templates whet level they should be at, though. @michellekondou, what do you think of that?

**Comment by aparkening** [comment]
As discussed in Slack, passing in parameters or otherwise adjusting the heading order programmatically is the best way to go.

**Comment by murtaugh** [comment]
@awkawk: would you have a moment to take a look at the home page on dev (http://dev.alistapart.com) and see what you think of the updated header hierarchy?

**Comment by awkawk** [comment]
I think that it looks great and meets the WCAG 2.0 AA bar, nice job.  If I was going to be _realllllly_ fussy I could comment that the "job board" heading 4 doesn't have an H3 parent as the "more from ALA" is an H2 and the previous H3 is "from the blog", but I don't think that this is going to affect the user experience.  We have the main issues perfect so the items that are from the blog has appropriate nesting for the headings and same goes for the main articles and the columns.

**Comment by michellekondou** [comment]
@aparkening @awkawk @murtaugh While working on updating the article template heading order, I run the page in https://gsnedders.html5.org/outliner and it seems our document outline could be improved. Below is a screenshot of the current article document outline (using the frameworks article as an example)

![ala-document-outline-original](https://cloud.githubusercontent.com/assets/1759784/11474364/614f0e24-9780-11e5-9be9-56cf20e65834.png)

and below is a screenshot of a document outline I ended up with after changing the heading and sectioning logic

![ala-document-outline-proposed](https://cloud.githubusercontent.com/assets/1759784/11474393/8e4dfe6c-9780-11e5-970c-4a7b01370beb.png)

What do you all think does that seem like a more meaningfull outline?
In this outline the logo is not an H1, the article heading is the pages H1 (referenced in this issue https://github.com/alistapart/AListApart/issues/326)
